### PR TITLE
Deprecate all addons except nginx-ingress and kubernetes-dashboard

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**: We want to get rid of most of the in-tree addons (long-term approach as described in #246), hence, we should deprecate all those which are any outdated, not needed any more, or complicating our extensibility epic (#308).

**Special notes for your reviewer**: This change required that we remove the old hibernation logic (min=max=0 for all worker pools) and only rely on the `.spec.hibernation` field (see #350 for further details).

**Release note**:

```action user
:warning: The deprecated option to trigger shoot cluster hibernation is no longer possible (setting `autoScalerMin=autoScalerMax=0` for all worker pools). The only way to hibernate a shoot cluster is setting `.spec.hibernation.enabled=true` as of now. Please adapt your scripts and change all existing shoot cluster specification accordingly before upgrading Gardener.
```
```noteworthy user
:warning: All addons other than `kubernetes-dashboard` and `nginx-ingress` are now deprecated. The cluster-autoscaler addon is automatically enabled if at least one worker pool specifies `max>min`. Please find more individual recommendations/explanations in the [example shoot resource](https://github.com/gardener/gardener/blob/master/example/90-shoot-aws.yaml). 
```

### CODE REVIEW

**Comment by bergerx** [comment]
I'm seeing these are the ones to be deprecated:
- ClusterAutoscaler
- Heapster
- Kube2IAM
- KubeLego
- Monocular

I'm more concerned about `ClusterAutoscaler` in partular, does this PR means there won't be any autoscaling function since its not mentioned in the description, #370, #246  and one of the addons listed in current shoot spec? Is the expectation from gardener users to deploy it theirself?

**Comment by rfranzke** [comment]
Sorry, the text was not clear enough. Please see https://github.com/gardener/gardener/pull/471/files#diff-821b13a69f022fe2f64512368125e741R101: The autoscaler is still part of Gardener core but it is no longer needed to activate/enable it separately as Gardener can detect whether it is needed or not itself.
I've updated the release note accordingly.

**Comment by luisdavim** [comment]
and the autoscaler is not using heapster? is it using kube-metrics?

**Comment by rfranzke** [comment]
The cluster-autoscaler has never used Heapster. It does not require an addiitonal component for identifying whether it needs to scale.
The horizontal pod autoscaler was using Heapster, but we have deprecated/removed Heapster already some weeks ago (#241). Since then we are deploying the Kubernetes metrics-server which is now used in all shoots for providing metrics for nodes and pods.

File: pkg/apis/garden/validation/validation_test.go [hunk 1]
```diff
@@ -2758,39 +2754,23 @@ var _ = Describe("validation", func() {
...
-				}
 
 				errorList := ValidateShoot(shoot)
 
-				Expect(len(errorList)).To(Equal(1))
-				Expect(*errorList[0]).To(MatchFields(IgnoreExtras, Fields{
-					"Type":  Equal(field.ErrorTypeForbidden),
-					"Field": Equal(fmt.Sprintf("spec.cloud.%s.workers[0].autoScalerMin/autoScalerMax", fldPath)),
-				}))
+				Expect(len(errorList)).To(Equal(0))
```
Comment (adracus) [inline pkg/apis/garden/validation/validation_test.go]: How about `BeEmpty()`?

**Review by adracus — DISMISSED** [review]
Minor comment (that doesn't need to be fixed IMO), otherwise LGTM

**Review by timuthy — DISMISSED** [review]

**Pull request merged**